spmd/tensor/parallel/style.py

```diff
@@ -0,0 +1,82 @@
+# Copyright (c) Meta Platforms, Inc. and affiliates
+import torch
+from dataclasses import dataclass
+from typing import Callable, Union, Optional
+from spmd.tensor import (
+    DTensor,
+    Shard,
+    Replicate,
+    DeviceMesh,
+)
+
+
+@dataclass
+class ParallelStyle(object):
```

Comment:
Ideally you can split this one (abstract class) into a separate PR. But this is fine this time. 

Also can we extend ABC instead? (https://docs.python.org/3/library/abc.html)

Reply:
TIL Python's Abstract Base Classes.